Client/BraveSkus/BraveSkusManager.swift

```diff
@@ -0,0 +1,155 @@
+      return
+    }
+    
+    guard expirationDate < Date() else {
+      Logger.module.debug("Existing sku credential has not expired yet, no need to refresh it.")
+      return
+    }
+    
+    guard let manager = BraveSkusManager(isPrivateMode: isPrivate) else {
+      assert(isPrivate, "SkusServiceFactory failed to intialize in regular mode, something is wrong.")
+      return
+    }
+    
+    Logger.module.debug("Refreshing sku credential. Clearing old credential from persistence.")
+    
+    BraveVPN.clearSkusCredentials()
+    
+    manager.credentialSummary(for: domain) { completion in
+      Logger.module.debug("credentialSummary response")
+    }
+  }
+  
+  // MARK: - Handling SKU methods.
+  
+  func refreshOrder(for orderId: String, domain: String, resultJSON: @escaping (Any?) -> Void) {
+    sku.refreshOrder(domain, orderId: orderId) { completion in
+      do {
+        guard let data = completion.data(using: .utf8) else { return }
+        let json = try JSONSerialization.jsonObject(with: data, options: .fragmentsAllowed)
+        Logger.module.debug("refreshOrder json parsed successfully")
+        resultJSON(json)
+      } catch {
+        resultJSON(nil)
+        Logger.module.error("refrshOrder: Failed to decode json: \(error.localizedDescription)")
+      }
+    }
+  }
+  
+  func fetchOrderCredentials(for orderId: String, domain: String, resultCredential: @escaping (String) -> Void) {
+    sku.fetchOrderCredentials(domain, orderId: orderId) { completion in
+      Logger.module.debug("skus fetchOrderCredentials")
+      resultCredential(completion)
+    }
+  }
+  
+  func prepareCredentialsPresentation(for domain: String, path: String,
+                                      resultCredential: ((String) -> Void)?) {
+    Logger.module.debug("skus prepareCredentialsPresentation")
+    sku.prepareCredentialsPresentation(domain, path: path) { credential in
+      if !credential.isEmpty {
+        if let vpnCredential = BraveSkusWebHelper.fetchVPNCredential(credential, domain: domain) {
+          Preferences.VPN.skusCredential.value = credential
+          Preferences.VPN.skusCredentialDomain.value = domain
+          Preferences.VPN.expirationDate.value = vpnCredential.expirationDate
+          
+          BraveVPN.setCustomVPNCredential(vpnCredential)
+        }
+      } else {
+        Logger.module.debug("skus empty credential from prepareCredentialsPresentation call")
+      }
+      
+      resultCredential?(credential)
+    }
+  }
+  
+  func credentialSummary(for domain: String, resultJSON: @escaping (Any?) -> Void) {
+    sku.credentialSummary(domain) { [weak self] completion in
+      do {
+        Logger.module.debug("skus credentialSummary")
+        
+        guard let data = completion.data(using: .utf8) else { return }
+        let json = try JSONSerialization.jsonObject(with: data, options: .fragmentsAllowed)
+        
+        resultJSON(json)
```

Comment:
Does the order of this matter? Should it be called before we do the remaining calls? Should it be called if the JSONDecoder below throws? etc.

Reply:
it's only for the website to render correct things, remaining code is unrelated and happens on our devices only. I will double check with @bsclifton

Reply:
The `credential_summary` call as called from the website is async. We should already have `.then()` handlers in place for when the response is returned. For example, when testing with JS console open, I usually do:
```
await window.chrome.braveSkus.credential_summary('vpn.brave.software')
```

Reply:
nm the above; I get the question now - yeah we resolve the promise as the last step in the C++ side of things:
https://github.com/brave/brave-core/blob/62dc0282726f959d4693c1dc8126f7d595df845d/components/skus/renderer/skus_js_handler.cc#L278-L316

Reply:
thanks, fixed here https://github.com/brave/brave-ios/pull/6397/commits/653e3d4ddf56b06bf80f94b682fe0de99607c6df